addons/mail/static/src/discuss/pinned_messages_panel.xml

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+    <t t-name="mail.PinnedMessagesPanel" owl="1">
+        <div class="o-mail-PinnedMessagePanel h-100 overflow-auto d-flex flex-column bg-100 position-relative" t-att-class="{
+            'flex-shrink-0 w-100 o-small': env.isSmall,
+        }">
+            <p class="o-mail-PinnedMessagePanel-title position-sticky top-0 py-3 mb-0 fw-bolder text-center text-uppercase text-700 bg-100">
+                Pinned messages
+            </p>
+            <t t-set="pinnedMessages" t-value="thread.pinnedMessages"/>
+            <div class="d-flex flex-column" t-att-class="{
+                'justify-content-center flex-grow-1': pinnedMessages.length === 0,
+            }" t-ref="message-list">
+                <div class="card m-3 mt-0"
+                    t-foreach="pinnedMessages"
+                    t-as="message"
+                    t-key="message.id"
+                    t-on-touchstart="() => this.onMessageTouchStart(message.id)"
+                    t-on-touchend="() => this.onMessageTouchEnd(message.id)"
+                >
+                    <div class="card-body">
+                        <div t-if="!env.isSmall" class="d-flex justify-content-end">
+                            <button class="o-mail-PinnedMessagePanel-jump btn rounded bg-400 text-light badge opacity-0 flex-shrink-0" t-on-click="() => this.onClickJump(message)">Jump</button>
+                            <div class="ms-2 text-700 opacity-75-hover cursor-pointer opacity-100" t-on-click="() => this.onClickUnpin(message)">
+                                <i class="d-block fa fa-times"/>
+                            </div>
+                        </div>
```

Comment:
I suppose you need some rights to unpin (pin option is not available for guests)